pkg/scheduler/ugm/manager_test.go

```diff
@@ -485,6 +505,528 @@ func TestUserGroupHeadroom(t *testing.T) {
+		{
+			name: "maxresources with a specific group limit and a wildcard group limit for a specific group user",
+			user: security.UserGroup{User: "user1", Groups: []string{"group1"}},
+			conf: createConfigWithLimits([]configs.Limit{
+				createLimit(nil, []string{"group1"}, mediumResource, 2),
+				createLimit(nil, []string{"*"}, largeResource, 2),
+			}),
+		},
+		{
+			name: "maxapplications with a specific group limit and a wildcard group limit for a specific group user",
+			user: security.UserGroup{User: "user1", Groups: []string{"group1"}},
+			conf: createConfigWithLimits([]configs.Limit{
+				createLimit(nil, []string{"group1"}, largeResource, 1),
+				createLimit(nil, []string{"*"}, largeResource, 2),
+			}),
+		},
+		// mixed user and group limit
+		{
+			name: "maxresources with user limit lower than group limit",
+			user: security.UserGroup{User: "user1", Groups: []string{"group1"}},
+			conf: createConfigWithLimits([]configs.Limit{
+				createLimit([]string{"user1"}, nil, mediumResource, 2),
+				createLimit(nil, []string{"group1"}, largeResource, 2),
+			}),
+		},
+		{
+			name: "maxapplications with user limit lower than group limit",
+			user: security.UserGroup{User: "user1", Groups: []string{"group1"}},
+			conf: createConfigWithLimits([]configs.Limit{
+				createLimit([]string{"user1"}, nil, largeResource, 1),
+				createLimit(nil, []string{"group1"}, largeResource, 2),
+			}),
+		},
+		{
+			name: "maxresources with gorup limit lower than user limit",
+			user: security.UserGroup{User: "user1", Groups: []string{"group1"}},
+			conf: createConfigWithLimits([]configs.Limit{
+				createLimit([]string{"user1"}, nil, largeResource, 2),
+				createLimit(nil, []string{"group1"}, mediumResource, 2),
+			}),
+		},
+		{
+			name: "maxapplications with group limit lower than user limit",
+			user: security.UserGroup{User: "user1", Groups: []string{"group1"}},
+			conf: createConfigWithLimits([]configs.Limit{
+				createLimit([]string{"user1"}, nil, largeResource, 2),
+				createLimit(nil, []string{"group1"}, largeResource, 1),
+			}),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// clear tracked resource
+			setupUGM()
+
+			manager := GetUserManager()
+			assert.NilError(t, manager.UpdateConfig(tc.conf.Queues[0], "root"))
+
+			usage, err := resources.NewResourceFromConf(mediumResource)
+			if err != nil {
+				t.Errorf("new resource create returned error or wrong resource: error %t, res %v", err, usage)
+			}
+
+			increased := manager.IncreaseTrackedResource(queuePathParent, TestApp1, usage, tc.user)
+			assert.Equal(t, increased, true, "unable to increase tracked resource: queuepath "+queuePathParent+", app "+TestApp1+", res "+usage.String())
+
+			increased = manager.IncreaseTrackedResource(queuePathParent, TestApp2, usage, tc.user)
+			assert.Equal(t, increased, false, "should not increase tracked resource: queuepath "+queuePathParent+", app "+TestApp2+", res "+usage.String())
+		})
```

Comment:
If we can assert user and group resource usage in addition to either yes or no, that gives clear picture on usage perspective as well.

Reply:
Add more assertions for this part. Thanks.